alicloud/resource_alicloud_alikafka_instance.go

```diff
@@ -0,0 +1,332 @@
+	d.Set("vswitch_id", object.VSwitchId)
+	d.Set("zone_id", object.ZoneId)
+
+	return nil
+}
+
+func resourceAlicloudAlikafkaInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*connectivity.AliyunClient)
+	alikafkaService := AlikafkaService{client}
+
+	// Process change instance name.
+	if d.HasChange("name") {
+		var name string
+		if v, ok := d.GetOk("name"); ok {
+			name = v.(string)
+		}
+		modifyInstanceNameReq := alikafka.CreateModifyInstanceNameRequest()
+		modifyInstanceNameReq.RegionId = client.RegionId
+		modifyInstanceNameReq.InstanceId = d.Id()
+		modifyInstanceNameReq.InstanceName = name
+
+		err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+			raw, err := alikafkaService.client.WithAlikafkaClient(func(alikafkaClient *alikafka.Client) (interface{}, error) {
+				return alikafkaClient.ModifyInstanceName(modifyInstanceNameReq)
+			})
+			if err != nil {
+				if IsExceptedErrors(err, []string{AlikafkaThrottlingUser}) {
+					time.Sleep(10 * time.Second)
+					return resource.RetryableError(err)
+				}
+				return resource.NonRetryableError(err)
+			}
+			addDebug(modifyInstanceNameReq.GetActionName(), raw, modifyInstanceNameReq.RpcRequest, modifyInstanceNameReq)
+			return nil
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), modifyInstanceNameReq.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+	}
+
+	attributeUpdate := false
+
+	upgradeReq := alikafka.CreateUpgradePostPayOrderRequest()
+	upgradeReq.RegionId = client.RegionId
+	upgradeReq.InstanceId = d.Id()
+	upgradeReq.TopicQuota = requests.NewInteger(d.Get("topic_quota").(int))
+	upgradeReq.DiskSize = requests.NewInteger(d.Get("disk_size").(int))
+	upgradeReq.IoMax = requests.NewInteger(d.Get("io_max").(int))
+
+	if d.HasChange("topic_quota") || d.HasChange("disk_size") || d.HasChange("io_max") {
+		attributeUpdate = true
+	}
+	eipMax := 0
+	if v, ok := d.GetOk("eip_max"); ok {
+		eipMax = v.(int)
+	}
+	if d.HasChange("eip_max") {
+
+		if v, ok := d.GetOk("eip_max"); ok {
+			eipMax = v.(int)
+		}
+		upgradeReq.EipMax = requests.NewInteger(eipMax)
+		attributeUpdate = true
+	}
+
+	if attributeUpdate {
+		err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+			raw, err := alikafkaService.client.WithAlikafkaClient(func(alikafkaClient *alikafka.Client) (interface{}, error) {
+				return alikafkaClient.UpgradePostPayOrder(upgradeReq)
+			})
+			if err != nil {
+				if IsExceptedErrors(err, []string{AlikafkaThrottlingUser}) {
+					time.Sleep(10 * time.Second)
+					return resource.RetryableError(err)
+				}
+				return resource.NonRetryableError(err)
+			}
+			addDebug(upgradeReq.GetActionName(), raw, upgradeReq.RpcRequest, upgradeReq)
+			return nil
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), upgradeReq.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+	}
+
+	err := alikafkaService.WaitForAlikafkaInstanceUpdated(d.Id(), d.Get("topic_quota").(int),
```

Comment:
Please use timeouts to do that, like https://github.com/terraform-providers/terraform-provider-alicloud/blob/master/alicloud/resource_alicloud_instance.go#L643